x-pack/platform/plugins/shared/ml/public/application/jobs/components/job_details_flyout/job_details_flyout.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiText,
+  EuiTitle,
+  useGeneratedHtmlId,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import type { CombinedJobWithStats } from '../../../../../common/types/anomaly_detection_jobs';
+import { useJobDetailFlyout } from './job_details_flyout_context';
+import { useMlApi } from '../../../contexts/kibana';
+import { JobDetails } from '../../jobs_list/components/job_details';
+import { loadFullJob } from '../../jobs_list/components/utils';
+import { useToastNotificationService } from '../../../services/toast_notification_service';
+
+export const JobDetailsFlyout = () => {
+  const {
+    isFlyoutOpen,
+    setIsFlyoutOpen,
+    activeJobId: jobId,
+    setActiveJobId,
+  } = useJobDetailFlyout();
+  const flyoutTitleId = useGeneratedHtmlId({
+    prefix: 'jobDetailsFlyout',
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [jobDetails, setJobDetails] = useState<CombinedJobWithStats | null>(null);
+  const mlApi = useMlApi();
+  const { displayErrorToast } = useToastNotificationService();
+
+  useEffect(() => {
+    let mounted = true;
+    const fetchJobDetails = async () => {
+      if (jobId) {
+        setIsLoading(true);
+        try {
+          const job = await loadFullJob(mlApi, jobId);
+          if (mounted && job) {
+            setJobDetails(job);
+          }
+        } catch (error) {
+          displayErrorToast(
+            error,
+            i18n.translate('xpack.ml.jobDetailsFlyout.errorFetchingJobDetails', {
+              defaultMessage: 'Error fetching job details',
+            })
+          );
+        } finally {
+          setIsLoading(false);
+        }
+      }
+    };
+    fetchJobDetails();
+    return () => {
+      mounted = false;
+    };
+  }, [jobId, mlApi, displayErrorToast]);
+
+  if (!jobId) {
+    return null;
+  }
+
+  return isFlyoutOpen ? (
+    <EuiFlyout
+      data-test-subj="jobDetailsFlyout"
+      type="overlay"
+      size="m"
+      ownFocus={false}
+      onClose={() => {
+        setIsFlyoutOpen(false);
+        setActiveJobId(null);
+      }}
+      aria-labelledby={flyoutTitleId}
+    >
+      <EuiFlyoutHeader hasBorder data-test-subj={`jobDetailsFlyout-${jobId}`}>
```

Comment:
Will you be adding a link to view the job in the Job management page into the header as in the designs?

Not sure what to use for the text. `Manage job` ?  Would need to change that depending on the user's permissions to edit jobs.

<img width="574" alt="Screenshot 2025-01-21 at 16 55 34" src="https://github.com/user-attachments/assets/6811cea3-0fa4-41b5-874e-e1b6cd84315f" />

Reply:
@peteharverson For this PR, do we want to use `Open jobs list` or `Manage job`? And in this context that would open the ML plugin's Anomaly detection job list page right?

Reply:
`Open jobs list` looks good.